tools/lsp/common/rename_component.rs

```diff
@@ -648,67 +648,43 @@ mod tests {
 
     use crate::common::test;
     use crate::common::text_edit;
-    use crate::preview;
-
-    /// Find the identifier that belongs to a struct of the given `name` in the `document`
-    fn find_struct_identifiers(
-        document: &syntax_nodes::Document,
-        name: &str,
-    ) -> Vec<syntax_nodes::DeclaredIdentifier> {
-        let mut result = vec![];
-        let name = i_slint_compiler::parser::normalize_identifier(name);
-
-        for el in document.ExportsList() {
-            for st in el.StructDeclaration() {
-                let identifier = st.DeclaredIdentifier();
-                if i_slint_compiler::parser::normalize_identifier(&identifier.text().to_string())
-                    == name
-                {
-                    result.push(identifier);
-                }
-            }
-        }
-
-        for st in document.StructDeclaration() {
-            let identifier = st.DeclaredIdentifier();
-            if i_slint_compiler::parser::normalize_identifier(&identifier.text().to_string())
-                == name
-            {
-                result.push(identifier);
-            }
-        }
-
-        result.sort_by_key(|i| i.text_range().start());
-        result
-    }
 
-    /// Find the identifier that belongs to a struct of the given `name` in the `document`
-    fn find_enum_identifiers(
-        document: &syntax_nodes::Document,
-        name: &str,
-    ) -> Vec<syntax_nodes::DeclaredIdentifier> {
-        let mut result = vec![];
-
-        for el in document.ExportsList() {
-            for en in el.EnumDeclaration() {
-                let identifier = en.DeclaredIdentifier();
-                if i_slint_compiler::parser::normalize_identifier(&identifier.text().to_string())
-                    == name
-                {
-                    result.push(identifier);
-                }
-            }
-        }
-
-        for en in document.EnumDeclaration() {
-            let identifier = en.DeclaredIdentifier();
-            if identifier.text() == name {
-                result.push(identifier);
+    #[track_caller]
+    fn find_node_by_comment(
+        document_cache: &common::DocumentCache,
+        document_path: &Path,
+        suffix: &str,
+    ) -> i_slint_compiler::parser::SyntaxNode {
+        let document = document_cache.get_document_by_path(document_path).unwrap();
+        let document = document.node.as_ref().unwrap();
+        let comment = document
+            .descendants_with_tokens()
```

Comment:
What I'm usually doing in the other tests is to just call find() on the source string to get the offset, and then find the token from the offset

Reply:
Yeap, that makes sense :-) I'll change that.

Reply:
Done!